tests/sparseml/pytorch/sparsification/quantization/test_quantize.py

```diff
@@ -0,0 +1,100 @@
+    QuantizationArgs,
+    QuantizationScheme,
+)
+
+
+@pytest.mark.parametrize(
+    "quantization_args_dict,target_quantization_args",
+    [
+        (
+            dict(num_bits=8, symmetric=False),
+            QuantizationArgs(num_bits=8, symmetric=False),
+        ),
+        (
+            dict(num_bits=4, symmetric=True),
+            QuantizationArgs(num_bits=4, symmetric=True),
+        ),
+        (
+            dict(num_bits=8, kwargs=dict(reduce_rage=True)),
+            QuantizationArgs(num_bits=8, kwargs=dict(reduce_rage=True)),
+        ),
+    ],
+)
+def test_quantization_args_from_dict(quantization_args_dict, target_quantization_args):
+    loaded_quantization_args = QuantizationArgs.parse_obj(quantization_args_dict)
+    assert isinstance(loaded_quantization_args, QuantizationArgs)
+    assert loaded_quantization_args == target_quantization_args
+
+
+@pytest.mark.parametrize(
+    "quantization_scheme_dict,target_quantization_scheme",
+    [
+        (
+            dict(input_activations=None, weights=None, output_activations=None),
+            QuantizationScheme(
+                input_activations=None, weights=None, output_activations=None
+            ),
+        ),
+        (
+            dict(
+                input_activations=dict(num_bits=8, symmetric=False),
+                weights=dict(num_bits=4, symmetric=True),
+                output_activations=dict(num_bits=8, kwargs=dict(reduce_rage=True)),
+            ),
+            QuantizationScheme(
+                input_activations=QuantizationArgs(num_bits=8, symmetric=False),
+                weights=QuantizationArgs(num_bits=4, symmetric=True),
+                output_activations=QuantizationArgs(
+                    num_bits=8, kwargs=dict(reduce_rage=True)
+                ),
+            ),
+        ),
+    ],
+)
+def test_quantization_scheme_from_dict(
+    quantization_scheme_dict, target_quantization_scheme
+):
+    loaded_quantization_scheme = QuantizationScheme.parse_obj(quantization_scheme_dict)
+    assert isinstance(loaded_quantization_scheme, QuantizationScheme)
+
+    def _assert_none_or_is_quant_args(val):
+        assert val is None or isinstance(val, QuantizationArgs)
+
+    _assert_none_or_is_quant_args(loaded_quantization_scheme.input_activations)
+    _assert_none_or_is_quant_args(loaded_quantization_scheme.weights)
+    _assert_none_or_is_quant_args(loaded_quantization_scheme.output_activations)
+
+    assert loaded_quantization_scheme == target_quantization_scheme
+
+
+@pytest.mark.parametrize(
+    "qunatization_args,target_quant_min,target_quant_max",
+    [
+        (QuantizationArgs(num_bits=8, symmetric=False), -128, 127),
+        (QuantizationArgs(num_bits=4, symmetric=True), -8, 7),
+    ],
+)
+def test_quantization_args_get_observer(
+    qunatization_args, target_quant_min, target_quant_max
+):
+    observer = qunatization_args.get_observer()
+    assert hasattr(observer, "with_args")
```

Comment:
Is this really necessary? What does this line check?

Reply:
type check isn't great here because a partial is technically returned. this checks that the partial has expected format